src/ansible_compat/runtime.py

```diff
@@ -352,6 +352,20 @@ def prepare_environment(
             return
 
         if os.path.exists("galaxy.yml"):
+            if destination:
```

Comment:
If the `destination` is None, the issue you want to work around also occur.

Do we need to consider the first default ansible_collection path(`~/.ansible/collections/`)?

Reply:
Sorted the none. I am not sure yet about the default. I preferred to enable this logic only for the case when the destination is specified, to limit the risks.